pages/index.tsx

```diff
@@ -1,150 +1,156 @@
-import Head from 'next/head'
-import Image from 'next/image'
-import { Inter } from '@next/font/google'
+import Head from "next/head";
+import Image from "next/image";
+import { Inter } from "@next/font/google";
 
+import products from "../db/products.json";
+import { useEffect, useState } from "react";
+import { Product } from "../db/models/store";
 
-const inter = Inter({ subsets: ['latin'] })
+
+
+const inter = Inter({ subsets: ["latin"] });
 
 export default function Home() {
-    return (
-        <>
-            <Head>
-                <title>Create Next App</title>
-                <meta name="description" content="Generated by create next app" />
-                <meta name="viewport" content="width=device-width, initial-scale=1" />
-                <link rel="icon" href="/favicon.ico" />
-                <link href="https://cdn.jsdelivr.net/npm/bootstrap@5.3.0-alpha1/dist/css/bootstrap.min.css" rel="stylesheet" integrity="sha384-GLhlTQ8iRABdZLl6O3oVMWSktQOp6b7In1Zl3/Jr59b6EGGoI1aFkw7cmDA6j6gD" crossOrigin="anonymous" />
-                <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/font-awesome/6.2.1/css/all.min.css" integrity="sha512-MV7K8+y+gLIBoVD59lQIYicR65iaqukzvf/nwasF0nqhPay5w/9lJmVM2hMDcnK1OnMGCdVK+iQrJ7lzPJQd1w==" crossOrigin="anonymous" referrerPolicy="no-referrer" />
-            </Head>
-            <main>
-                <header>
-                    <nav className="navbar bg-body-tertiary">
-                        <div className="container-fluid">
-                            <a className="navbar-brand">Shop <i className="fa fa-shopping-bag" aria-hidden="true"></i>
-                            </a>
-                            <form className="d-flex" role="search">
-                                <input className="form-control me-2" type="search" placeholder="Search" aria-label="Search" />
-                            </form>
+  const [products, setProducts] = useState<Product[]>([]);
```

Comment:
nit: should we be using async await over the `.then` promise syntax?

Reply:
Where?

Reply:
I guess you mean:

```
const fetchProducts = () => {
    fetch("/api/store")
      .then((response) => response.json())
      .then((data) => {
        if (data.result) {
          setProducts(data.result);
        }
      })
      .catch((e) => {
        console.log(`Error: ${e}`);
      });
  };

  const searchQuery = () => {
    fetch(`/api/store/search?query=${encodeURI(searchInput)}`, {
      method: "GET",
    })
      .then((response) => response.json())
      .then((data) => {
        if (data.result) {
          setProducts(data.result);
        }
      });
  };
```

Reply:
Since it's not deprecated, we should use it.

Reply:
> I guess you mean:
> 
> ```
> const fetchProducts = () => {
>     fetch("/api/store")
>       .then((response) => response.json())
>       .then((data) => {
>         if (data.result) {
>           setProducts(data.result);
>         }
>       })
>       .catch((e) => {
>         console.log(`Error: ${e}`);
>       });
>   };
> 
>   const searchQuery = () => {
>     fetch(`/api/store/search?query=${encodeURI(searchInput)}`, {
>       method: "GET",
>     })
>       .then((response) => response.json())
>       .then((data) => {
>         if (data.result) {
>           setProducts(data.result);
>         }
>       });
>   };
> ```

Yes, I'm asking if we should update this to use `async` and `await` for easier to read code?

Reply:
Let's do that.